DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc

```diff
@@ -249,209 +286,256 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
+          if (indexLastFilterInTriggerEvent < triggerEventSize) {
+            if (this->skipModuleByType(hltConfigProvider_.moduleType(labelLastFilterInPath))) {
+              continue;
             }
+            break;
           }
         }
+        // number of modules in the path
+        unsigned int const nModulesInPath = hltConfigProvider_.size(pathIndex);
+        LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::analyze]         "
+                                        << "-> selected indexLastFilterInPath = " << indexLastFilterInPath
+                                        << " (HLTConfigProvider::size(" << pathIndex << ") = " << nModulesInPath << ")";
+        if (indexLastFilterInPath >= nModulesInPath) {
+          edm::LogError("HLTFiltersDQMonitor")
+              << " selected index (" << indexLastFilterInPath << ") for last filter of path \"" << iPathName
+              << "\" is inconsistent with number of modules in the Path (" << nModulesInPath << ")";
+          continue;
+        }
+        // store decision of previous filter
+        bool previousFilterAccept(true);
+        for (size_t modIdx = 0; modIdx < nModulesInPath; ++modIdx) {
+          // each filter-bin is filled, with a 0 or 1, only when all previous filters in the Path have passed
+          if (not previousFilterAccept) {
+            break;
+          }
+          // consider only selected EDFilter modules
+          auto const& moduleLabel = hltConfigProvider_.moduleLabel(pathIndex, modIdx);
+          if (this->skipModuleByEDMType(hltConfigProvider_.moduleEDMType(moduleLabel)) or
+              this->skipModuleByType(hltConfigProvider_.moduleType(moduleLabel))) {
+            continue;
+          }
+          // index of module in this Path [0,nModulesInPath)
+          unsigned int const slotModule = hltConfigProvider_.moduleIndex(pathIndex, moduleLabel);
+          bool filterAccept = false;
+          if (slotModule < indexLastFilterInPath) {
+            filterAccept = true;
+          } else if (slotModule == indexLastFilterInPath) {
+            filterAccept = pathAccept;
+          }
+          LogTrace("HLTFiltersDQMonitor")
+              << "[HLTFiltersDQMonitor::analyze]         "
+              << "HLTConfigProvider::moduleLabel(" << pathIndex << ", " << modIdx << ") = \"" << moduleLabel
+              << "\", HLTConfigProvider::moduleIndex(" << pathIndex << ", \"" << moduleLabel << "\") = " << slotModule
+              << ", filterAccept = " << filterAccept << ", previousFilterAccept = " << previousFilterAccept;
+
+          auto const ibinKey = idset + "." + iPathName + "." + moduleLabel;
+          if (binIndexMap_.find(ibinKey) == binIndexMap_.end()) {
+            throw cms::Exception("HLTFiltersDQMonitorInvalidBinLabel")
+                << "invalid key for bin-index map (name of Module bin in MonitorElement of Path): \"" << ibinKey
+                << "\"";
+          }
+          auto const ibin = binIndexMap_[ibinKey];
```

Comment:
Also here storing the returned iterator from `binIndexMap_.find(ibinKey)` in a variable and de-referencing it would be more efficient.